### PR TITLE
Integrate VCR, apply censors

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Integrate EasyVCR
- Cassettes now properly censored
- Unit tests improved
- Improve how merging two objects works, accounting for inherited classes

NOTE: Cassettes currently only match by method and URL, awaiting improvements to EasyVCR.

NOTE 2: Because of how our library is designed, the VCR has to be injected into our main code (not just the test suite), making EasyVCR a main dependency.

# Testing

- Cassettes recorded, work on replay

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -140,7 +146,6 @@
                     <source>8</source>
                     <doclint>-html</doclint>
                     <quiet>true</quiet>
-                    <additionalparam>-Xdoclint:none</additionalparam>
```
Comment (Justintime50) [inline pom.xml:143]: What's this for?

File: src/main/java/com/easypost/net/EasyPostResource.java [hunk 2]
```diff
@@ -121,19 +123,19 @@ static Map<String, String> getHeaders(String apiKey) {
         return headers;
     }
 
-    private static javax.net.ssl.HttpsURLConnection createEasyPostConnection(final String url, final String apiKey)
-            throws IOException {
-        URL easypostURL = null;
+    private static javax.net.ssl.HttpsURLConnection createEasyPostConnection(final String url, final String apiKey,
+                                                                             final String method) throws IOException {
```
Comment (Justintime50) [inline src/main/java/com/easypost/net/EasyPostResource.java:127]: Is this a checkstyle thing? Seems like a super odd indentation?

File: src/test/java/com/easypost/BatchTest.java [hunk 3]
```diff
@@ -130,11 +154,21 @@ public void testBuy() throws EasyPostException {
...
-        Batch batchWithScanForm = globalBatch.createScanForm();
+    @Test
+    public void testCreateScanForm() throws EasyPostException, InterruptedException {
+        _vcr.setUpTest("create_scanform");
+
+        Batch batch = createOneCallBuyBatch();
+        batch = batch.buy();
+
+        if (_vcr.isRecording()) {
+            Thread.sleep(10000); // Wait enough time for processing
```
Comment (Justintime50) [inline src/test/java/com/easypost/BatchTest.java:165]: Nit: 5 seconds is sufficient

File: src/test/java/com/easypost/BatchTest.java [hunk 4]
```diff
@@ -154,6 +190,7 @@ public void testAddRemoveShipment() throws EasyPostException {
         shipmentData.add(shipment);
         params.put("shipments", shipmentData);
 
+        // TODO: We shouldn't be requiring users to put the list of shipments in a dictionary first
```
Comment (Justintime50) [inline src/test/java/com/easypost/BatchTest.java:193]: Let's make an internal task instead and remove the TODO. This will easily get lost here forever.

File: src/test/java/com/easypost/BatchTest.java [hunk 5]
```diff
@@ -169,29 +206,24 @@ public void testAddRemoveShipment() throws EasyPostException {
...
-                Batch batchWithLabel = globalBatch.label(params);
+        batch = batch.buy();
 
-                // We can't assert anything meaningful here because the label gets queued for generation and may not be immediately available
-                assertEquals(1, batchWithLabel.getNumShipments().intValue());
-                assertTrue(batchWithLabel instanceof Batch);
-                break;
-            }
+        if (_vcr.isRecording()) {
+            Thread.sleep(10000); // Wait enough time for processing
```
Comment (Justintime50) [inline src/test/java/com/easypost/BatchTest.java:217]: Nit: 5 seconds is sufficient.

File: src/test/java/com/easypost/BatchTest.java [hunk 6]
```diff
@@ -169,29 +206,24 @@ public void testAddRemoveShipment() throws EasyPostException {
...
         }
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("file_format", "ZPL");
+
+        // TODO: We shouldn't be requiring users to put the format in a dictionary first
```
Comment (Justintime50) [inline src/test/java/com/easypost/BatchTest.java:223]: Ticket instead please

File: src/test/java/com/easypost/CarrierAccountTest.java [hunk 7]
```diff
@@ -102,11 +129,16 @@ public void testUpdate() throws EasyPostException {
...
     public void testDelete() throws EasyPostException {
-        globalCarrierAccount.delete();
+        _vcr.setUpTest("delete");
 
-        assertTrue(globalCarrierAccount instanceof CarrierAccount);
+        CarrierAccount carrierAccount = createBasicCarrierAccount();
+
+        carrierAccount.delete();
+
+        // TODO: endpoint should return boolean to evaluate
```
Comment (Justintime50) [inline src/test/java/com/easypost/CarrierAccountTest.java:139]: Indeed, though we have a ticket for this so no need for a TODO in a test.

File: src/test/java/com/easypost/CarrierAccountTest.java [hunk 8]
```diff
@@ -102,11 +129,16 @@ public void testUpdate() throws EasyPostException {
...
+
+        testCarrierAccountId = null; // skip deletion cleanup
```
Comment (Justintime50) [inline src/test/java/com/easypost/CarrierAccountTest.java:141]: Based on what you mentioned, this shouldn't be needed anymore?

File: src/test/java/com/easypost/ErrorTest.java [hunk 9]
```diff
@@ -2,22 +2,24 @@
...
-
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.BeforeAll;
-import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class ErrorTest {
+    // TODO: Need more unit tests for this class.
```
Comment (Justintime50) [inline src/test/java/com/easypost/ErrorTest.java:11]: Granted, no need for the TODO. We have a task for this

File: src/test/java/com/easypost/EventTest.java [hunk 10]
```diff
@@ -53,10 +62,16 @@ public void testAll() throws EasyPostException {
...
+        List<Event> events = eventCollection.getEvents();
+        Event event = events.get(0);
+
+        Event retrievedEvent = Event.retrieve(event.getId());
 
-        assertTrue(retrievedEvent instanceof Event);
-        assertTrue(retrievedEvent.getId().startsWith("evt_"));
-        assertEquals(globalEvents.getEvents().get(0).getId(), retrievedEvent.getId());
+        assertInstanceOf(Event.class, retrievedEvent);
+        // Must compare IDs because other elements of objects may be different
```
Comment (Justintime50) [inline src/test/java/com/easypost/EventTest.java:74]: I'm unsure that this is true of events, can we double check?

File: src/test/java/com/easypost/Fixture.java [hunk 11]
```diff
@@ -73,6 +76,14 @@ public static String reportDate() {
         return "2022-04-12";
     }
 
+    /**
+     * Prefix for report type.
+     * @return report type prefix
+     */
+    public static String reportIdPrefix() {
+        return "shprep_";
```
Comment (Justintime50) [inline src/test/java/com/easypost/Fixture.java:84]: Fixtures are meant to be reusable data that we pass to the API specifically. 

I'd rather we either had this bit in the report test file itself or simply hard-coded the assertion (in this case it should be fine since it's what is always expected and shouldn't change.)

File: src/test/java/com/easypost/InsuranceTest.java [hunk 12]
```diff
@@ -58,11 +58,15 @@ public void testCreate() throws EasyPostException {
...
+
+        Insurance insurance = createBasicInsurance();
+
+        Insurance retrievedInsurance = Insurance.retrieve(insurance.getId());
 
-        assertTrue(retrievedInsurance instanceof Insurance);
-        assertTrue(retrievedInsurance.getId().startsWith("ins_"));
-        assertEquals(globalInsurance.getId(), retrievedInsurance.getId());
+        assertInstanceOf(Insurance.class, insurance);
+        // Must compare IDs since other elements of object may be different
```
Comment (Justintime50) [inline src/test/java/com/easypost/InsuranceTest.java:68]: May be able to compare the whole thing?

File: src/test/java/com/easypost/OrderTest.java [hunk 13]
```diff
@@ -83,16 +100,20 @@ public void testGetRates() throws EasyPostException {
...
+        _vcr.setUpTest("buy");
 
+        Order order = createBasicOrder();
+
+        Map<String, Object> params = new HashMap<>();
         params.put("carrier", Fixture.usps());
         params.put("service", Fixture.uspsService());
 
-        globalOrder.buy(params);
+        // TODO: We shouldn't require the user to wrap these parameters into a dictionary, should just be two parameters
```
Comment (Justintime50) [inline src/test/java/com/easypost/OrderTest.java:111]: No TODO

File: src/test/java/com/easypost/PickupTest.java [hunk 14]
```diff
@@ -72,16 +86,19 @@ public void testRetrieve() throws EasyPostException {
...
+        _vcr.setUpTest("buy");
+
+        Pickup pickup = createBasicPickup();
 
+        Map<String, Object> params = new HashMap<>();
         params.put("carrier", Fixture.usps());
         params.put("service", Fixture.pickupService());
 
-        Pickup boughtPickup = globalPickup.buy(params);
+        // TODO: We shouldn't be requiring the end user to wrap these parameters in a dictionary
```
Comment (Justintime50) [inline src/test/java/com/easypost/PickupTest.java:98]: No TODO

File: src/test/java/com/easypost/ScanFormTest.java [hunk 15]
```diff
@@ -1,41 +1,50 @@
...
         List<Shipment> shipments = new ArrayList<>();
-        Map<String, Object> params = new HashMap<>();
-
         shipments.add(shipment);
+
+        Map<String, Object> params = new HashMap<>();
         params.put("shipments", shipments);
 
-        globalScanForm = ScanForm.create(params);
+        // TODO: We shouldn't require the end-user to wrap these parameters in a dictionary.
```
Comment (Justintime50) [inline src/test/java/com/easypost/ScanFormTest.java:45]: No TODO

File: src/test/java/com/easypost/ShipmentTest.java [hunk 16]
```diff
@@ -150,16 +186,19 @@ public void testConvertLabel() throws EasyPostException {
...
         // Set to 0 so USPS doesn't insure this automatically and we can insure the
         // shipment manually.
+        shipmentData.put("insurance", 0);
+
+        Map<String, Object> insuranceData = new HashMap<>();
+        insuranceData.put("amount", "100");
 
         Shipment shipment = Shipment.create(shipmentData);
 
+        // TODO: We shouldn't require the end-user to wrap this parameter in a dictionary.
```
Comment (Justintime50) [inline src/test/java/com/easypost/ShipmentTest.java:201]: No TODO

File: src/test/java/com/easypost/TestUtils.java [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+
+public abstract class TestUtils {
+    private static final String API_KEY_FAILED_TO_PULL = "couldnotpullapikey";
+
+    private static final String CASSETTES_FOLDER = "cassettes";
+
+    private static final java.util.List<String> HEADER_CENSORS = new ArrayList<String>() {{
+        add("Authorization");
+        add("X-Client-User-Agent");
+        add("User-Agent");
```
Comment (Justintime50) [inline src/test/java/com/easypost/TestUtils.java:22]: Should be sorted

File: src/test/java/com/easypost/TestUtils.java [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+    }};
+
+    private static final List<String> QUERY_CENSORS = new ArrayList<String>() {{
+        add("card[number]");
+        add("card[cvc]");
+    }};
+
+    private static final List<String> BODY_CENSORS = new ArrayList<String>() {{
+        add("api_keys");
+        // add("children");
```
Comment (Justintime50) [inline src/test/java/com/easypost/TestUtils.java:32]: Let's remove this completely

File: src/test/java/com/easypost/TestUtils.java [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+        add("client_ip");
+        add("credentials");
+        add("key");
+        add("keys");
+        add("phone_number");
+        add("phone");
+        add("test_credentials");
+        // add("created_at"); TODO: Add as ignore in body in future
```
Comment (Justintime50) [inline src/test/java/com/easypost/TestUtils.java:40]: Let's say `why` we need to do this here

File: src/test/java/com/easypost/TestUtils.java [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+        // add("updated_at");
+    }};
+
+    private static String getSourceFileDirectory() {
```
Comment (Justintime50) [inline src/test/java/com/easypost/TestUtils.java:44]: Missing a docstring?

File: src/test/java/com/easypost/TestUtils.java [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+         */
+        public VCR(String testCassettesFolder, String apiKey) {
+            AdvancedSettings advancedSettings = new AdvancedSettings();
+            advancedSettings.matchRules = MatchRules.regular(); // match by method, url
+            // TODO: Change to strict when VCR updated with ignore aspect
+            advancedSettings.censors =
+                    new Censors("<REDACTED>").hideHeaders(HEADER_CENSORS).hideQueryParameters(QUERY_CENSORS)
+                            .hideBodyParameters(BODY_CENSORS);
+            advancedSettings.simulateDelay = false;
+            advancedSettings.manualDelay = 0;
```
Comment (Justintime50) [inline src/test/java/com/easypost/TestUtils.java:110]: Do we need these or will VCR set these defaults for us? If so, let's remove them here

File: src/test/java/com/easypost/TrackerTest.java [hunk 22]
```diff
@@ -1,33 +1,43 @@
...
+    /**
+     * Create a tracker.
+     */
+    private static Tracker createBasicTracker() throws EasyPostException {
+        Map<String, Object> params = new HashMap<>();
+        params.put("carrier", Fixture.usps());
         params.put("tracking_code", "EZ1000000001");
 
-        globalTracker = Tracker.create(params);
+        // TODO: We shouldn't require end-users to wrap these parameters in a dictionary.
```
Comment (Justintime50) [inline src/test/java/com/easypost/TrackerTest.java:39]: No TODO

File: src/test/java/com/easypost/UserTest.java [hunk 23]
```diff
@@ -80,16 +121,21 @@ public void testRetrieveMe() throws EasyPostException {
...
+        User user = createUser();
+
+        String testName = "New Name";
         Map<String, Object> params = new HashMap<>();
 
-        params.put("phone", testPhone);
+        params.put("name", testName);
 
-        globalUser.update(params);
+        // TODO: Failing because name not overriding on merge?
```
Comment (Justintime50) [inline src/test/java/com/easypost/UserTest.java:133]: This has been fixed, remove the TODO

File: src/test/java/com/easypost/UserTest.java [hunk 24]
```diff
@@ -98,10 +144,16 @@ public void testUpdate() throws EasyPostException {
...
-              // deleting them as we cannot replace the deleted ones easily.
     public void testDelete() throws EasyPostException {
-        globalUser.delete();
+        _vcr.setUpTest("delete");
+
+        User user = createUser();
+
+        user.delete();
+
+        // TODO: endpoint should return boolean to evaluate
```
Comment (Justintime50) [inline src/test/java/com/easypost/UserTest.java:154]: No TODO

File: src/test/java/com/easypost/WebhookTest.java [hunk 25]
```diff
@@ -102,9 +112,13 @@ public void testAll() throws EasyPostException {
...
     @Test
-    @Disabled
     public void testUpdate() throws EasyPostException {
-        // Cannot be easily tested - requires a disabled webhook.
+        _vcr.setUpTest("update");
+
+        Webhook webhook = createBasicWebhook();
+
+        webhook.update();
+        // TODO: We should call this something more intuitive in the future, since it doesn't work like the other Update function
```
Comment (Justintime50) [inline src/test/java/com/easypost/WebhookTest.java:121]: We need to test this, I suggest we test to see if it's an instance of. I understand it's not great, but it's better than nothing. This is one of those unfortunate things we can't really test well.

The TODO can go. Although the naming is kind of weird, it does indeed update it.

File: src/test/java/com/easypost/WebhookTest.java [hunk 26]
```diff
@@ -113,12 +127,16 @@ public void testUpdate() throws EasyPostException {
...
-        // doesn't fail.
-        globalWebhook.delete();
+        _vcr.setUpTest("delete");
+
+        Webhook webhook = createBasicWebhook();
+        Webhook retrievedWebhook = Webhook.retrieve(webhook.getId());
+
+        retrievedWebhook.delete();
+
+        // TODO: endpoint should return boolean to evaluate
```
Comment (Justintime50) [inline src/test/java/com/easypost/WebhookTest.java:138]: No TODO

File: src/test/java/com/easypost/beta/EndShipperTest.java [hunk 27]
```diff
@@ -77,24 +93,27 @@ public void testAll() throws EasyPostException {
...
+        updateParams.put("name", testName);
         updateParams.put("company", "EasyPost");
         updateParams.put("street1", "388 Townsend St");
         updateParams.put("street2", "Apt 20");
         updateParams.put("city", "San Francisco");
         updateParams.put("state", "CA");
         updateParams.put("zip", "94107");
         updateParams.put("country", "US");
-        updateParams.put("phone", "9999999999");
+        updateParams.put("phone", "1234567890");
```
Comment (Justintime50) [inline src/test/java/com/easypost/beta/EndShipperTest.java:110]: Eh, can leave. Only need to swap one thing

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Awesome work here. Most of these are removal of TODOs and a couple cleanup items.

File: pom.xml (hunk 1 above)
Comment (nwithan8) [reply pom.xml:143]: Maven is complaining about its inclusion

File: src/main/java/com/easypost/net/EasyPostResource.java (hunk 2 above)
Comment (nwithan8) [reply src/main/java/com/easypost/net/EasyPostResource.java:127]: Looks fine on my end, think it was a GitHub diff thing.

File: src/test/java/com/easypost/EventTest.java (hunk 10 above)
Comment (nwithan8) [reply src/test/java/com/easypost/EventTest.java:74]: Cannot currently do whole-object comparisons in Java.

File: src/test/java/com/easypost/InsuranceTest.java (hunk 12 above)
Comment (nwithan8) [reply src/test/java/com/easypost/InsuranceTest.java:68]: Cannot currently do whole-object comparisons in Java.

File: src/test/java/com/easypost/EventTest.java (hunk 10 above)
Comment (Justintime50) [reply src/test/java/com/easypost/EventTest.java]: Hmmm, why not? I thought we had a PR dedicated to just overcoming this problem.

**Review by Justintime50 — DISMISSED** [review]
Let's gooo!

File: src/main/java/com/easypost/model/PaymentMethod.java [hunk 28]
```diff
@@ -47,15 +47,6 @@ public String getId() {
         return id;
     }
 
-    /**
-     * Set ID of this PaymentMethod object.
-     *
-     * @param id ID of this PaymentMethod.
-     */
```
Comment (jchen293) [inline src/main/java/com/easypost/model/PaymentMethod.java:54]: I know these setters are not necessary, but why are you deleting them in this PR?

File: src/main/java/com/easypost/net/EasyPostResource.java [hunk 29]
```diff
@@ -192,33 +201,39 @@ private static javax.net.ssl.HttpsURLConnection writeBody(final javax.net.ssl.Ht
...
     private static javax.net.ssl.HttpsURLConnection createGetConnection(final String url, final String query,
                                                                         final String apiKey) throws IOException {
-        String getURL = String.format("%s?%s", url, query);
-        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(getURL, apiKey);
+        String getURL = url;
+        if (query != null) {
+            getURL = String.format("%s?%s", url, query);
+        }
+        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(getURL, apiKey, "GET");
         conn.setRequestMethod("GET");
```
Comment (jchen293) [inline src/main/java/com/easypost/net/EasyPostResource.java:209]: This line can be removed since we are passing in `method` in the parameter and we are setting the request method in the `createEasyPostConnection` function in line 165

File: src/main/java/com/easypost/net/EasyPostResource.java [hunk 30]
```diff
@@ -192,33 +201,39 @@ private static javax.net.ssl.HttpsURLConnection writeBody(final javax.net.ssl.Ht
...
         return conn;
     }
 
     private static javax.net.ssl.HttpsURLConnection createPostConnection(final String url, final JsonObject body,
                                                                          final String apiKey) throws IOException {
-        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(url, apiKey);
+        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(url, apiKey, "POST");
         conn.setRequestMethod("POST");
```
Comment (jchen293) [inline src/main/java/com/easypost/net/EasyPostResource.java:216]: Same here, this can be removed.

File: src/main/java/com/easypost/net/EasyPostResource.java [hunk 31]
```diff
@@ -192,33 +201,39 @@ private static javax.net.ssl.HttpsURLConnection writeBody(final javax.net.ssl.Ht
...
     private static javax.net.ssl.HttpsURLConnection createDeleteConnection(final String url, final String query,
                                                                            final String apiKey) throws IOException {
-        String deleteUrl = String.format("%s?%s", url, query);
-        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(deleteUrl, apiKey);
+        String deleteUrl = url;
+        if (query != null) {
+            deleteUrl = String.format("%s?%s", url, query);
+        }
+        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(deleteUrl, apiKey, "DELETE");
         conn.setRequestMethod("DELETE");
```
Comment (jchen293) [inline src/main/java/com/easypost/net/EasyPostResource.java:228]: Same here, this can be removed.

File: src/main/java/com/easypost/net/EasyPostResource.java [hunk 32]
```diff
@@ -192,33 +201,39 @@ private static javax.net.ssl.HttpsURLConnection writeBody(final javax.net.ssl.Ht
...
         return conn;
     }
 
     private static javax.net.ssl.HttpsURLConnection createPutConnection(final String url, final JsonObject body,
                                                                         final String apiKey) throws IOException {
-        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(url, apiKey);
+        javax.net.ssl.HttpsURLConnection conn = createEasyPostConnection(url, apiKey, "PUT");
         conn.setRequestMethod("PUT");
```
Comment (jchen293) [inline src/main/java/com/easypost/net/EasyPostResource.java:235]: Same here, this can be removed.

File: src/test/java/com/easypost/AddressTest.java [hunk 33]
```diff
@@ -105,18 +117,15 @@ public void testAll() throws EasyPostException{
...
         Map<String, Object> addressData = Fixture.incorrectAddressToVerify();
-        addressData.put("verify", verificationList);
 
         Address address = Address.create(addressData);
 
-        assertTrue(address instanceof Address);
+        assertInstanceOf(Address.class, address);
         assertTrue(address.getId().startsWith("adr_"));
         assertEquals("417 MONTGOMERY ST FL 5", address.getStreet1());
-        assertEquals("Invalid secondary information(Apt/Suite#)", address.getVerifications().getZip4().getErrors().get(0).getMessage());
```
Comment (jchen293) [inline src/test/java/com/easypost/AddressTest.java:119]: Why is this removed?

File: src/test/java/com/easypost/AddressTest.java [hunk 34]
```diff
@@ -70,11 +78,14 @@ public void testCreateVerifyStrict() throws EasyPostException {
...
+
+        Address address = createBasicAddress();
+        Address retrievedAddress = Address.retrieve(address.getId());
+
+        assertInstanceOf(Address.class, retrievedAddress);
+        assertEquals(address.getId(), retrievedAddress.getId());
 
-        assertTrue(retrievedAddress instanceof Address);
-        assertTrue(retrievedAddress.getId().startsWith("adr_"));
-        assertThat(globalAddress).usingRecursiveComparison().isEqualTo(retrievedAddress);
```
Comment (jchen293) [inline src/test/java/com/easypost/AddressTest.java:77]: Was this removed because we can no longer compare objects due to cassette scrubbing?

File: src/test/java/com/easypost/BatchTest.java [hunk 35]
```diff
@@ -169,29 +205,23 @@ public void testAddRemoveShipment() throws EasyPostException {
...
+        Batch batch = createOneCallBuyBatch();
 
-        while (true) {
-            globalBatch = globalBatch.refresh();
-            if ("postage_purchased".equals(globalBatch.getShipments().get(0).getBatchStatus())) {
-                Batch batchWithLabel = globalBatch.label(params);
+        batch = batch.buy();
 
-                // We can't assert anything meaningful here because the label gets queued for generation and may not be immediately available
-                assertEquals(1, batchWithLabel.getNumShipments().intValue());
```
Comment (jchen293) [inline src/test/java/com/easypost/BatchTest.java:191]: This was accidentally removed.

File: src/test/java/com/easypost/EasyPostTest.java [hunk 36]
```diff
@@ -26,6 +27,7 @@ public static void setup() {
      */
     @Test
     public void testConnectionTimeout() {
+
```
Comment (jchen293) [inline src/test/java/com/easypost/EasyPostTest.java:30]: nit: we don't need extra line here

File: src/test/java/com/easypost/PaymentMethodTest.java [hunk 37]
```diff
@@ -25,9 +28,10 @@ public static void setup() throws EasyPostException{
      */
     @Test
     public void testAll() throws EasyPostException {
-        PaymentMethod billing = PaymentMethod.all();
+        _vcr.setUpTest("all");
+
+        PaymentMethod paymentMethods = PaymentMethod.all();
 
-        assertTrue(billing.getPrimaryPaymentMethod() != null);
-        assertTrue(billing.getSecondaryPaymentMethod() != null);
```
Comment (jchen293) [inline src/test/java/com/easypost/PaymentMethodTest.java:31]: Why were these two assertations removed?

File: src/test/java/com/easypost/ShipmentTest.java (hunk 16 above)
Comment (jchen293) [reply src/test/java/com/easypost/ShipmentTest.java:201]: TODO comment is still here

File: src/test/java/com/easypost/AddressTest.java (hunk 33 above)
Comment (nwithan8) [reply src/test/java/com/easypost/AddressTest.java:119]: Unit test was made to match C#

File: src/test/java/com/easypost/AddressTest.java (hunk 34 above)
Comment (nwithan8) [reply src/test/java/com/easypost/AddressTest.java:77]: Unit test was made to match C#

File: src/test/java/com/easypost/BatchTest.java (hunk 35 above)
Comment (nwithan8) [reply src/test/java/com/easypost/BatchTest.java:191]: Unit test was made to match C#

File: src/test/java/com/easypost/PaymentMethodTest.java (hunk 37 above)
Comment (nwithan8) [reply src/test/java/com/easypost/PaymentMethodTest.java:31]: Weirdly enough, the API just isn't returning the data we need. May be VCR related, but for now, we don't have the data to do these assertions.

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]
